### ISSUE TITLE
layout_with_fr not producing proper layout

### ISSUE THREAD

**Issue by ecalvo68** [igraph/rigraph#110]
The layout_with_fr function is not working as it did in previous versions (layout.fruchterman.reingold). Estimates for the layout collapse to zero. The comparison between the same code in R 3.0 (upper graph) and the new code in R 3.2 (lower graph)
![original-salta](https://cloud.githubusercontent.com/assets/13910144/9419984/cb91d470-482e-11e5-8ea8-013ea4df6923.JPG)
Compare to this one
![new-salta](https://cloud.githubusercontent.com/assets/13910144/9419986/d15119fc-482e-11e5-9e5f-18515d64cb4f.JPG)

**Comment by gaborcsardi** [comment]
The previous version of the FR layout did not have much to do with the original FR paper, unfortunately. As for the messed up layout, can you please send a reproducible example?

**Comment by ecalvo68** [comment]
Hi Gabor,
 As you will see, the layout_with_fr() breaks down at 1500 edges. This happens in all examples I am running. The sample data (R workspace) can be downloaded from: 
"https://dl.dropboxusercontent.com/u/62912960/Salta%20PASO%20PROVINCIA-8%20PM.RData.RData"

#This code will produce two network graphs, one with 1499 edges and one with 1500. The first one #comes out perfect. The second one shows what happens with all networks I have tried using the #new version 1.0 when the network is relatively large

library(ggplot2)
library(igraph)

op <- par(mfcol=c(1,2))  
for(i in 1:2){
data <- cbind(author.paired,author.retweeted)
data<-data[1:1498+i,]
net <- graph.empty()
net <- add.vertices(net, length(unique(c(data))),name=as.character(unique(c(data))))
net <- add.edges(net, t(data))
summary(net) 

l <-layout.fruchterman.reingold(net) 
my.com.fast <- walktrap.community(net)

cent<-data.frame(bet=log(betweenness(net)+1),eig=log(evcent(net)$vector+1))
res<-lm(eig~bet,data=cent)$residuals
cent<-transform(cent,res=res)
ind<-degree(net, mode="in")
V(net)$size<-abs(res)
nodes<-as.vector(V(net))
nodes[which(abs(res)<.07)]<-NA
comps <- clusters(net)$membership
E(net)$color <- "beige" #rgb(.5,.8,.7,.4) #"aliceblue"
new.color<-data.frame(t(col2rgb(my.com.fast$membership+4)/255))
new.color<-rgb(new.color, alpha=.15)

plot.igraph(net,vertex.label=my.label2, layout=l, vertex.size=log(ind+1), vertex.label.color=1, vertex.color=new.color, vertex.frame.color=new.color, edge.width= .01, edge.arrow.size=.1, vertex.label.cex=1,  edge.curved=TRUE, vertex.label.dist=rnorm(length(ind),.1,.03))
}
par(op)

**Comment by gaborcsardi** [comment]
Thanks. Can you please try to make the example minimal? Is it important to have the colors? Why do I need ggplot2?

**Comment by ecalvo68** [comment]
#Sure. Here is a basic version. Again, all large networks are collapsing in the new version.Thanks!

library(igraph)
op <- par(mfcol=c(1,2))
for(i in 1:2){
data <- cbind(author.paired,author.retweeted)
data<-data[1:1498+i,]
net <- graph.empty()
net <- add.vertices(net, length(unique(c(data))),name=as.character(unique(c(data))))
net <- add.edges(net, t(data))
summary(net)
l <-layout.fruchterman.reingold(net) 
plot.igraph(net,layout=l, vertex.label="")
}
par(op)

**Comment by gaborcsardi** [comment]
I cannot run this:

```
Error in cbind(author.paired, author.retweeted) :
  object 'author.paired' not found
```

**Comment by ecalvo68** [comment]
Did you load the data? Is in:
https://dl.dropboxusercontent.com/u/62912960/Salta%20PASO%20PROVINCIA-8%20PM.RData.RData

I cannot post a large dataset here, that is why I posted the link.

**Comment by gaborcsardi** [comment]
Thanks!

Both networks have 1498 edges, in fact. `1:1498 + i` is not what you think it is, consider:

```r
1:5 + 1
#> [1] 2 3 4 5 6
```

Anyway, the layout is indeed bad for the second graph, so this seems to be a bug. Here is a complete reproducible example:

```r
load(url("https://dl.dropboxusercontent.com/u/62912960/Salta%20PASO%20PROVINCIA-8%20PM.RData.RData"))
library(igraph)
par(mfcol=c(1,2))
for (i in 1:2) {
  data <- cbind(author.paired, author.retweeted)
  data <- data[1:(1498 + i), ]
  net <- graph_from_data_frame(data)
  l <- layout_with_fr(net)
  plot(net, layout = l, vertex.label = "")
}
```

**Comment by ecalvo68** [comment]
Thank you for the reproducible example! Also let me thank you for a great package. I am running networks with 200K edges and 20 K nodes that look amazing.

**Comment by mdoellma** [comment]
I have run into the same issue with layout_with_fr. It will plot nicely if the number of vertices is <= 999, but will collapse points onto the x- and y-axes if the number of vertices is >= 1000 (as in ecalvo68's comment on Aug 21, 2015). I am working with igraph_1.0.1 in R. 

Is the bug mentioned by gaborcsardi on Aug 22, 2015 being fixed, or is layout_with_fr inherently unable to handle > 999 vertices? I can post data and code, if needed.

**Comment by mdoellma** [comment]
Apologies for posting before I checked the source code! In layout.R, I found the following comment: 

#' @param grid Character scalar, whether to use the faster, but less accurate
#' grid based implementation of the algorithm. By default (\dQuote{auto}), the
#' grid-based implementation is used if the graph has more than one thousand
#' vertices.

Setting the grid parameter to "nogrid" when calling layout_with_fr() for a graph with >= 1000 vertices fixes the problem.

**Comment by shawnpg** [comment]
Good thread. Was hitting a similar issue in a graph with 6k nodes, and setting grid="nogrid" in layout_with_fr let igraph handle it like a champ. Thanks for the great library!

**Comment by szhorvat** [comment]
The problems with the grid layout are now in the C core (see https://github.com/igraph/igraph/issues/2068). The fix will be part of the release 1.3.2.

@ntamas Can you please update rigraph to the latest C core master and adapt any of the unit tests if needed?

**Comment by ntamas** [comment]
C core updated.
